### ISSUE TITLE
`DoNotCallSuggester ` does not warn on always-throwing methods in nested classes

### ISSUE THREAD

**Issue by Chordrain** [google/error-prone#5224]
**Description**
`DoNotCallSuggester` correctly flags methods in top-level classes that always throw exceptions, suggesting the use of `@DoNotCall`. However, for methods inside nested classes, it currently does **not** trigger any warning.

---

**Steps to Reproduce**

1. Use Error Prone 2.41.0
2. Create the following code:

```java
final class Test {

    class NestedClass {

        // Expected: should trigger DoNotCallSuggester
        public void foo() {
            throw new RuntimeException();
        }
    }
}
```

3. Compile and run the Error Prone check.

---

**Actual Behavior**

* Error Prone **does not emit any warning** for the `foo()` method in the nested class.

**Expected Behavior**

* `DoNotCallSuggester` should detect that `NestedClass.foo()` always throws an exception and recommend annotating it with `@DoNotCall`, similar to how it handles top-level class methods.

---

**Additional Information**

* Example of a top-level class method that **does** trigger the warning:

```java
final class Test {
    // Triggers DoNotCallSuggester
    public void foo() {
        throw new RuntimeException();
    }
}
```

* Possible reason: the checker currently only scans top-level class methods and may skip methods inside nested classes.
* Impact: Always-throwing methods in nested classes may not be annotated, which could lead API users to encounter runtime exceptions without any compile-time warning.

**Comment by cpovirk** [comment]
The warning appears if you change `NestedClass` or `foo()` to be `final`, just as `Test` is `final`.

We don't warn for overrideable methods because they could be stub implementations like, for example, [`AbstractList.set`](https://github.com/openjdk/jdk/blob/7e738f0d906e574706a277fabbc2cc1df6f11f19/src/java.base/share/classes/java/util/AbstractList.java#L137), for which a call on a _subclass_ implementation might be fine.
